File: Frontend/tehtavaappfrontend/src/interfaces/models/SchoolGroup.ts
import { Student } from '../models/Student';

/**
 * Enhanced SchoolGroup interface with all required properties
 * This contains a superset of all group properties used in the application
 */
export interface SchoolGroup {
  id: string;
  name: string;
  description?: string;
  createdById?: string;
  createdByName?: string;
  isActive?: boolean;
  memberCount?: number;
  studentCount?: number;
  courseCount?: number;
  courseId?: string;
  students?: Student[];
  studentEnrollments?: StudentGroupEnrollment[];
  hasCourse?: boolean;
  createdAt: string | Date;
  updatedAt?: string | Date;
}

/**
 * Interface for group enrollment
 */
export interface StudentGroupEnrollment {
  id: string;
  groupId: string;
  studentId: string;
  enrollmentDate: string | Date;
  status: string;
  progress: number;
  student?: Student;
  group?: SchoolGroup;
} 

File: Backend/TehtavaApp.API/Models/SchoolGroup.cs
namespace TehtavaApp.API.Models;

public class SchoolGroup
{
    public int Id { get; set; }
    public string Name { get; set; }
    public string Description { get; set; }
    public string CreatedById { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation properties
    public virtual ApplicationUser CreatedBy { get; set; }
    public virtual ICollection<StudentGroupEnrollment> StudentEnrollments { get; set; }
    public virtual ICollection<Course> Courses { get; set; }

    public SchoolGroup()
    {
        StudentEnrollments = new HashSet<StudentGroupEnrollment>();
        Courses = new HashSet<Course>();
    }
}
